From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Herman van Hazendonk <github.com@herrie.org>
Date: Fri, 15 Aug 2026 00:00:00 +0000
Subject: [PATCH] libhybris-egl-server: fix QtOpenGL move and server-buffer
 protocol

The libhybris server-buffer plugin does not build, then does not link:

    libhybriseglserverbufferintegration.cpp:9:10: fatal error:
    QtGui/QOpenGLTexture: No such file or directory

    wayland-libhybris-egl-server-buffer-protocol.c.o:(.data.rel+0x10):
    undefined reference to `qt_server_buffer_interface'

Two updates this plugin missed, both of which its siblings in
hardwareintegration/ already carry:

  - QOpenGLTexture moved from QtGui to QtOpenGL in Qt 6.0. drm-egl-server,
    dmabuf-server, shm-emulation-server and vulkan-server all include
    <QtOpenGL/QOpenGLTexture> and link Qt::OpenGL; this one still asked QtGui.

  - libhybris-egl-server-buffer.xml only declares the qt_server_buffer
    interface ("new_id" arg); the interface itself is defined in
    server-buffer-extension.xml. The other three plugins list both xml files in
    their qt6_generate_wayland_protocol_client_sources() call, so they get
    qt_server_buffer_interface generated; this one lists only its own, leaving
    the symbol undefined.

Nothing noticed because the plugin only builds when libhybris is present, which
upstream CI does not have. It surfaced for LuneOS when Qt 6.10 moved the
QtWayland client into qtbase.

Upstream-Status: Pending
Signed-off-by: Herman van Hazendonk <github.com@herrie.org>
---
 .../libhybris-egl-server/CMakeLists.txt                       | 2 ++
 .../libhybriseglserverbufferintegration.cpp                   | 2 +-
 2 files changed, 3 insertions(+), 1 deletion(-)

--- a/src/plugins/platforms/wayland/plugins/hardwareintegration/libhybris-egl-server/CMakeLists.txt
+++ b/src/plugins/platforms/wayland/plugins/hardwareintegration/libhybris-egl-server/CMakeLists.txt
@@ -17,6 +17,7 @@
         EGL::EGL
         Qt::Core
         Qt::Gui
+        Qt::OpenGL
         Qt::WaylandClientPrivate
         Wayland::Client
     QT_LICENSE_ID QT_COMMERCIAL_OR_LGPL3
@@ -26,6 +27,7 @@
     PRIVATE_CODE
     FILES
         ${CMAKE_CURRENT_SOURCE_DIR}/../../../../../../3rdparty/wayland/extensions/libhybris-egl-server-buffer.xml
+        ${CMAKE_CURRENT_SOURCE_DIR}/../../../../../../3rdparty/wayland/extensions/server-buffer-extension.xml
 )
 
 #### Keys ignored in scope 1:.:.:libhybris-egl-server.pro:<TRUE>:
--- a/src/plugins/platforms/wayland/plugins/hardwareintegration/libhybris-egl-server/libhybriseglserverbufferintegration.cpp
+++ b/src/plugins/platforms/wayland/plugins/hardwareintegration/libhybris-egl-server/libhybriseglserverbufferintegration.cpp
@@ -6,7 +6,7 @@
 #include <QtWaylandClient/private/qwaylanddisplay_p.h>
 #include <QDebug>
 #include <QtGui/QOpenGLContext>
-#include <QtGui/QOpenGLTexture>
+#include <QtOpenGL/QOpenGLTexture>
 #include <hybris/eglplatformcommon/hybris_nativebufferext.h>
 
 #include <EGL/egl.h>
